InventoryIndexer/Indexer/Stock/IndexDataProviderByStockId.php

```diff
@@ -22,31 +23,48 @@ class IndexDataProviderByStockId
     private $resourceConnection;
 
     /**
-     * @var SelectBuilder
+     * @var SelectBuilderInterface[]
      */
-    private $selectBuilder;
+    private $selectBuilders;
 
     /**
      * @param ResourceConnection $resourceConnection
-     * @param SelectBuilder $selectBuilder
+     * @param SelectBuilderInterface[] $selectBuilders
+     * @throws LocalizedException
      */
     public function __construct(
         ResourceConnection $resourceConnection,
-        SelectBuilder $selectBuilder
+        array $selectBuilders
```

Comment:
Could you please explain why this parameter should be array? I see only one select builder used.

Reply:
Not one,  two of them defined as arguments (defined in di.xml)
Magento\InventoryConfigurableProductIndexer\Indexer\SelectBuilder
Magento\InventoryIndexer\Indexer\SelectBuilder

The purpose to add extension point to the DataProvider